runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -5180,19 +5180,18 @@ class INTERPRETER_CLASS
 	yieldContinuationImpl(REGISTER_ARGS_LIST)
 	{
 		VM_BytecodeAction rc = EXECUTE_BYTECODE;
+		jint res = 0;
+
 		buildInternalNativeStackFrame(REGISTER_ARGS);
 		updateVMStruct(REGISTER_ARGS);
 
 		/* store the current Continuation state and swap to carrier thread stack */
-		yieldContinuation(_currentThread);
+		res = yieldContinuation(_currentThread);
 
 		VMStructHasBeenUpdated(REGISTER_ARGS);
 		restoreInternalNativeStackFrame(REGISTER_ARGS);
 
-		/* its going to return as if it were returning from continuation.enterImpl()
-		 * so we need to push the boolean return val
-		 */
-		returnSingleFromINL(REGISTER_ARGS, JNI_FALSE, 1);
+		returnSingleFromINL(REGISTER_ARGS, res, 1);
```

Comment:
There have to be a check on `res` here to to determine if the slots value should be `1` or `0`
for res = 0, we are returning to `enterImpl` which is non-static method with 0 arg, so slot = 1,
for res = 1, we are returning back to `yieldImpl` which is static + 0 arg, hence slot need to be 0 in that case

Reply:
I tried the above suggestion, and I also added some debugging code. The issue still persists. In `test_VirtualthreadYieldResume`, the first `enterCont*` is correct; then, the thread also yields correctly; but, it does not resume correctly since the test output says `results[i] == 1` at the end; passing condition: `results[i] == 2`.
```suggestion
		returnSingleFromINL(REGISTER_ARGS, res, (res == 0) ? 1 : 0);
```

https://github.com/eclipse-openj9/openj9/blob/e1ce5d219fbd59e940c30eb4c7a64a0c2970e51e/test/functional/Java19andUp/src/org/openj9/test/jep425/VirtualThreadTests.java#L65-L70

Reply:
Worked after accounting for ` also have to be updated to return 0 instead of JNI_TRUE since the signature of yieldImpl changed`.